### PR TITLE
refactor: files{} -> challengeFiles[], and key -> fileKey

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Related to #39719

As usual, my PR went a bit further than hoped 😅 
- [x] Renames cases of `files` -> `challengeFiles`
- [x] Renames cases of `key` (`challengeFile.key`) -> `fileKey`
- [x] Miscellaneous fixes thrown in, as needed and pointed out by TS and in my escapade
- [x] Converts `challengeFiles` to an array, to match the current database schema
- [x] Cleans up (kind-of) `prop-types` - eventually, this file needs to be renamed, and probably moved (once client no longer uses PropTypes package)
- [x] Removes a lot of unnecessary TS-ignoring comments (my OCD kicked in)

### CODE REVIEW

File: api-server/src/common/models/user.json [hunk 1]
```diff
@@ -276,9 +276,7 @@
       "default": {}
     },
     "donationEmails": {
-      "type": [
-        "string"
-      ]
+      "type": ["string"]
```
Comment (raisedadead) [inline api-server/src/common/models/user.json:279]: Is this just a cosmetic change, or did we lose some changes during the rebasing?

Comment (nhcarrigan) [reply api-server/src/common/models/user.json:279]: This is a prettier change. I got the same change while working on #42663

Comment (raisedadead) [reply api-server/src/common/models/user.json:279]: Awesome, thanks for confirming.

**Comment by raisedadead** [comment]
What happened to the commits here? Did we not fix it?

**Comment by ShaunSHamilton** [comment]
> What happened to the commits here? Did we not fix it?

@raisedadead I am not sure. I was under the impression this was fixed, but I could have done something incorrect, when adding the latest 2 commits.

**Comment by raisedadead** [comment]
Can I recommend starting with a fresh branch off of main (unfortunately a new PR), and cherry-picking only the commits that you need? I am sure your local branch is a bit stale.

**Pull request closed without merging**